lib/spack/spack/environment.py

```diff
@@ -621,6 +621,10 @@ def __init__(self, path, init_file=None, with_view=None):
                     self._set_user_specs_from_lockfile()
                 else:
                     self._read_manifest(f, raw_yaml=default_manifest_yaml)
+
+                if hasattr(f, 'name') and f.name.endswith('.yaml'):
+                    init_file_dir = os.path.abspath(os.path.dirname(f.name))
+                    self._rewrite_relative_paths_on_relocation(init_file_dir)
```

Comment:
This needs a comment to explain why it's checking the file for a name.

Additionally, there should probably be an option to determine whether this is done. I'm picturing two workflows.

1. I create an environment and use it for a while, it points to repos where I have work and I want to keep working in those repos, but now I want the environment to be managed by Spack so that it can use reference counting to ensure I don't uninstall anything I need. I want the paths to be rewritten when I run `spack env create foo /path/to/spack.yaml`
2. I'm given an environment, and I want my copy of it to be managed by Spack. So the first thing I do before downloading any source code is run `spack env create foo /path/to/spack.yaml`. I don't want the paths to be rewritten in this case, the next thing I'm going to do is tell the new environment to download its own copies.

Reply:
Right, another thing is you'd probably would not want to run `spack create my_env ./spack.yaml` when `spack.yaml` has a relative develop path that escapes the environment directory `../xyz`.

I just defaulted to expanding the develop path and added `--keep-relative` to force spack to copy paths verbatim.

The reason is I like to store a `spack.yaml` env file inside my git repo for either CI or development with e.g. `path: ../..`